File: Two_Pointers/container_with_most_water.cpp
#include <bits/stdc++.h>

using namespace std;

int maxArea(vector<int>& height) {
    int left = 0;
    int right = height.size() - 1;

    int max_height = 0;

    while(left < right){
        if(min(height[left], height[right]) * (right - left) > max_height){
            max_height = min(height[left], height[right]) * (right - left);
        }
        if(height[left] < height[right]){
            left++;
        } else {
            right--;
        }
    }

    return max_height;
}

int main(){
    vector<int> height = {1,8,6,2,5,4,8,3,7};
    cout << maxArea(height);
}

File: Two_Pointers/container_with_most_water.py
class Solution:


    def maxArea(self, height: List[int]) -> int:
        max_count = 0
        left = 0
        right = len(height) - 1

        def calc_height(left, right):
            return min(height[left], height[right]) * (right - left)

        while left < right:
            curr_count = calc_height(left, right)
            if curr_count > max_count:
                max_count = curr_count

            if height[left] < height[right]:
                left += 1
            else:
                right -= 1

        return max_count 